### PR TITLE
added asserts to regression tests

### PR DESCRIPTION
**Pull request by staabm**
as requested in https://github.com/phpstan/phpstan-src/pull/1596#pullrequestreview-1064299237

### CODE REVIEW

File: tests/PHPStan/Analyser/data/bug-7417.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+function doFoo() {
+	$extensions = readThing();
+	$extensions['theme']['test_basetheme'] = 0;
+// This is the important part of the test. Themes are ordered alphabetically
+// in core.extension so this will come before it's base theme.
+	$extensions['theme']['test_subtheme'] = 0;
+	$extensions['theme']['test_subsubtheme'] = 0;
+	assertType('non-empty-array', $extensions); // could be more precise
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/bug-7417.php]: room for improvement, I think

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
You're adding the same classes that already exist in different files. This can cause big headaches later.

Please just reuse the files from tests/PHPStan/Rules and add these assertType calls in them. Thanks.

**Comment by staabm** [comment]
done 👍

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**